skfem/element/discrete_field.py

```diff
@@ -4,86 +4,73 @@
 from numpy import ndarray
 
 
-class DiscreteField(NamedTuple):
-    """A function defined at the global quadrature points.
-
-    Created using :meth:`~skfem.element.Element.gbasis` or
-    :meth:`~skfem.assembly.CellBasis.interpolate`.
-
-    """
-
-    value: ndarray = np.array([0])  # zero field
-    grad: Optional[ndarray] = None
-    div: Optional[ndarray] = None
-    curl: Optional[ndarray] = None
-    hess: Optional[ndarray] = None
-    grad3: Optional[ndarray] = None
-    grad4: Optional[ndarray] = None
-    grad5: Optional[ndarray] = None
-    grad6: Optional[ndarray] = None
-
-    def __array__(self):
-        return self.value
-
-    def __add__(self, other):
-        if isinstance(other, DiscreteField):
-            return self.value + other.value
-        return self.value + other
-
-    def __sub__(self, other):
-        if isinstance(other, DiscreteField):
-            return self.value - other.value
-        return self.value - other
-
-    def __mul__(self, other):
-        if isinstance(other, DiscreteField):
-            return self.value * other.value
-        return self.value * other
-
-    def __truediv__(self, other):
-        if isinstance(other, DiscreteField):
-            return self.value / other.value
-        return self.value / other
-
-    def __pow__(self, other):
-        if isinstance(other, DiscreteField):
-            return self.value ** other.value
-        return self.value ** other
-
-    def __neg__(self):
-        return -self.value
-
-    def __radd__(self, other):
-        return self.__add__(other)
-
-    def __rsub__(self, other):
-        return other - self.value
-
-    def __rmul__(self, other):
-        return self.__mul__(other)
-
-    def __rtruediv__(self, other):
-        return other / self.value
-
-    def __rpow__(self, other):
-        return other ** self.value
-
-    def _split(self):
-        """Split all components based on their first dimension."""
-        return [DiscreteField(*[f[i] for f in self if f is not None])
-                for i in range(self.value.shape[0])]
+class DiscreteField(ndarray):
+
+    def __new__(cls,
+                value=np.array([0]),
+                grad=None,
+                div=None,
+                curl=None,
+                hess=None,
+                grad3=None,
+                grad4=None,
+                grad5=None,
+                grad6=None):
+        obj = np.asarray(value).view(cls)
```

Comment:
This line is wrong. I tried to fix and push, but duh, this failed obviously. 

Anyway, the correct code should use None in the function def and then

```python
        if value is None:
            obj = np.asarray([0]).view(cls)
        else:
            obj = np.asarray(value).view(cls)
```

See [here](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) for why, if it matters.

Reply:
Thanks!